qiskit_nature/operators/second_quantization/qubit_converter.py

```diff
@@ -52,26 +52,31 @@ class QubitConverter:
     def __init__(self,
                  mapper: QubitMapper,
                  two_qubit_reduction: bool = False,
-                 auto_symmetry_reduction: bool = False,
-                 z2symmetry_reduction: Optional[List[int]] = None):
+                 z2symmetry_reduction: Optional[Union[str, List[int]]] = None):
         """
 
         Args:
             mapper: A mapper instance used to convert second quantized to qubit operators
             two_qubit_reduction: Whether to carry out two qubit reduction when possible
-            auto_symmetry_reduction: Whether to perform automatic symmetry reduction. This requires
-                the `sector_locator` argument of `convert` to be specified. This option will also be
-                overwritten if `z2symmetry_reduction` is not `None`.
-            z2symmetry_reduction: An optional sector definition so the desired operator, from
-               the tapered set, containing the problem solution is returned. This is a list of
-               -1 and 1's to define the sector, where the list size is the number of symmetries
-               of the main operator.
+            z2symmetry_reduction: Indicates whether a z2 symmetry reduction should be applied to
+                resulting qubit operators that are computed. For each symmetry detected the operator
+                will be split into two where each requires one qubit less for computation. So for
+                example 3 symmetries will split the original operator into 8 new operators each
+                requiring 3 less qubits. Now only one of these operators will have the ground state
+                and be the correct symmetry sector needed for the ground state. Setting 'auto' will
+                use an automatic computation of the correct sector. If the sector is known
+              from other experiments with the z2symmetry logic, then the tapering values of that
+                sector can be provided (a list of int of values -1, and 1). The default is None
+                meaning no symmetry reduction is done. Note that dipole and other operators such as
+                spin, num particles etc. are also symmetry reduced according to the symmetries found
+                in the main operator if this operator commutes with the main operator symmetry. If
+                it does not then the operator will be discarded since no meaningful measurement can
+                take place.
```

Comment:
This is really around the convert_match from the perspective here that if you do symmetry reduction on `convert` then `convert_match` will do the same symmetry tapering/reduction as long as the operator commutes with the symmetry. If not then qubit operator that would normally be output (so at that position in the List when a List is supplied) will be set to None. Of course this applies to any operators such as dipoles etc that are passed by the algorithm into this convertor.

If we want to avoid updating this at present, due to another build cycle we can, but I think it would be better to describe it as what it does from this components perspective relating it to the overall behavior in the stack like I outlined above.

Reply:
I don't fully understand what you want to say here. I admit just copying over this docstring from Aqua..
Are you just suggesting to clarify that additional operators are dealt with by `convert_match`?

Reply:
In Aqua or course this was all done as part of Transformation (formerly core.Hamiltonian) where that itself built out the dipoles etc in that same module. Here the QubitConvertor does nothing aside from what it is asked to do. It has no notion of dipoles or anything. Just some operator that it is asked to `convert`, an possibly taper (do symmetry reduction) and then other operators as passed later to `convert_match` where it will convert them the same way, but instead of finding symmetry it just matches the symmetry of the main operator, if you will, ie. it checks the operator commutes with symmetry. I think I am saying the text, and no you say the source, is more about the stack behavior when combined say with an ElectronicStructureProblem than about the QubitConvertor behavior. Thats what I tried to express above about convert and convert_match that I think should be the way things are described wrt putting None for non-commuting operators - now one give an example saying these operator could be dipoles etc if that seems helpful for a user.